beacon-chain/forkchoice/doubly-linked-tree/proposer_boost_test.go

```diff
@@ -429,14 +454,18 @@ func TestForkChoice_BoostProposerRoot(t *testing.T) {
 			store: &Store{},
 		}
 		// Genesis set to 1 slot ago + X where X > attesting interval.
-		genesis := time.Now().Add(-time.Duration(cfg.SecondsPerSlot) * time.Second)
-		attestingInterval := time.Duration(cfg.SecondsPerSlot / cfg.IntervalsPerSlot)
-		greaterThanAttestingInterval := attestingInterval + 100*time.Millisecond
-		genesis = genesis.Add(-greaterThanAttestingInterval * time.Second)
-		blockRoot := [32]byte{'A'}
+		attestingInterval := time.Duration(cfg.SecondsPerSlot/cfg.IntervalsPerSlot) * time.Second
+		greaterThanAttestingInterval := attestingInterval + time.Second
```

Comment:
why the jump from 100 milliseconds to 1000 milliseconds?

Reply:
Hey @potuz, so the proposer boost function actually operates in INTERVALS_PER_SLOT units where one interval is defined as SECONDS_PER_SLOT / 3. The logic is as follows in semi-go pseudocode:
```go
secondsPerSlot := SECONDS_PER_SLOT
isBeforeAttestingInterval := args.SecondsIntoSlot < secondsPerSlot/ INTERVALS_PER_SLOT

if args.CurrentSlot == args.BlockSlot && isBeforeAttestingInterval { 
  BOOST
}
```
as a function argument, all we care about is that we are earlier than the first interval in a slot, which is defined as unit of seconds. As such, I decided to pass in an argument of `secondsIntoSlot uint64` into the function, which makes our life easier in testing and in usage. The unit test above only worked in milliseconds, so I had to amend to seconds to make it work. The logic is the same, but now we are 1 second greater than the attesting interval rather than 100ms. We are still testing the same behavior